tasks/main.yml

```diff
@@ -1 +1,31 @@
 # role tasks
+
+- name: Install necessary packages
+  package:
+    name: parted
+    state: present
+  become: "{{ disk_wipe_become }}"
+  become_user: "{{ disk_wipe_become_user }}"
+
+- name: Clear the disk signatures
+  become: "{{ disk_wipe_become }}"
+  become_user: "{{ disk_wipe_become_user }}"
+  command: "wipefs --all --force {{ item }}"
+  register: signatures_before
+  changed_when: '"offset" in signatures_before.stdout'
+  with_items: "{{ disk_wipe_disks }}"
+
+- name: Re-read partion tables
+  become: "{{ disk_wipe_become }}"
+  become_user: "{{ disk_wipe_become_user }}"
+  command: "partprobe {{ item }}"
+  changed_when: false
+  with_items: "{{ disk_wipe_disks }}"
+
+- name: Erase the disk data
```

Comment:
In this case we are running `shred` to write zeroes to a specific part of the disk.  The name of this task should reflect that.  

Unlike `wipefs`, I believe that partition deletion using `shred` would probably require notifying the kernel about changes to partitions.  

It might make sense to run `shred` before `wipefs`, which may allow us to drop the `parted` dependency.

Reply:
Advice taken and role simplified.  run shred first and then wipefs as it will call the BLKRRPART ioctl when it has erased a partition-table signature to inform the kernel about the change.